src/core/curand_wrapper.hpp

```diff
@@ -0,0 +1,13 @@
+#ifndef CURAND_WRAPPER_HPP
+#define CURAND_WRAPPER_HPP
+
+#if defined(__HIPCC__) // AMD or Nvidia via HIP
+
+#include <hiprand_kernel.h>
```

Comment:
I think you need _hiprand/hiprand_kernel.h_. It still won't work as the function's name is different, but I can take care of that.